soc/arm/nxp_imx/rt/mpu_regions.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <devicetree.h>
+#include "../../common/cortex_m/arm_mpu_mem_cfg.h"
```

Comment:
I see you are not the first to need this header file.

Shall we do zephyr_include_directories(.) from soc/arm/common/cortex_m/CMakeLists.txt ?

Or perhaps even better move the header to

zephyr/include/arch/arm/aarch32/mpu/

```
sebo@sebo:~/ncs/bootloader/mcuboot$ ack arm_mpu_mem_cfg.h ~/ncs/
/home/sebo/ncs/zephyr/soc/arm/common/cortex_m/arm_mpu_regions.c
10:#include "arm_mpu_mem_cfg.h"

/home/sebo/ncs/zephyr/soc/arm/nuvoton_npcx/npcx7/mpu_regions.c
7:#include "../../common/cortex_m/arm_mpu_mem_cfg.h"

/home/sebo/ncs/zephyr/soc/arm/st_stm32/stm32h7/mpu_regions.c
8:#include "../../common/cortex_m/arm_mpu_mem_cfg.h"

```

Reply:
Also shameless plug for #41223 about this

Reply:
@SebastianBoe I agree that we should take one of those approaches with the file, but should it be something addressed by this PR, or a separate PR specifically calling out the issue and fixing it?

Reply:
A separate PR would be best.